votingworks/vxsuite
# Save All Ballot PDFs

## Overview
Closes #2146. Introduces `Export Ballot PDFs` button on the Ballots tab that allows an election administrator to export all the PDFs of a given type & mode at once. The export also includes a `.csv` metadata file that they could edit and use to send a job to their printer.

## Demo Video or Screenshot
[export-ballot-pdfs-rev2.webm](https://user-images.githubusercontent.com/37960853/181384963-c036bbbf-70b5-407d-8fd0-ffabb29d32b1.webm)

## Testing Plan
- Unit testing of new utility function
- Button / Modal Component tests

There are two additional things that would be nice to test but the effort is out of the scope of this PR, unless someone has a better suggestion:
1. Check that the ballot PDFs exported as part of this function match the printed ballots. Ideally we would be generating the ballots for each in a shared place, but that would require a deep dive into changing how we print / PDFize our documents in general. And very ideally, we would be checking that these PDFs work are accepted by our scanning service.
2. Check that the zip archive contains the file entries it is supposed to. Doing this would require an actual file download, meaning Cypress, but Cypress can't use the feature because AFAIK we can't mock USB drive detection in Cypress.

## Checklist
- [x] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [ ] I have added JSDoc comments to any newly introduced exports

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates
